SwrveSDK/SDK/Talk/SwrveTrigger.m

```diff
@@ -0,0 +1,53 @@
+
+#import "SwrveTrigger.h"
+#import "SwrveTriggerCondition.h"
+
+@implementation SwrveTrigger
+
+@synthesize eventName = _eventName;
+@synthesize conditions = _conditions;
+
+- (id) initWithDictionary:(NSDictionary *)dictionary {
+    self = [super init];
+    if(self) {
+        _eventName = [[dictionary objectForKey:kTriggerEventNameKey] lowercaseString];
+        _conditions = [self produceConditionsFromDictionary:dictionary];
+    }
+    
+    return self;
+}
+
+- (NSArray *) produceConditionsFromDictionary:(NSDictionary *) dictionary {
+    
+    NSString *triggerOperator = [dictionary objectForKey:@"op"];
```

Comment:
I'd vote for fail out, and don't add this condition.

If, for some reason bad data came down, it's better not to add this condition at all, rather than potentially add it without any trigger conditions.  In the song example scenario we've been using, if someone puts a "song.played" event in, with 20 ands, if it gets in without being parsed correctly, it would trigger every "song.played" event, rather than the specific condition it was designed for.